### PR TITLE
Integrate Configurator External Service

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Can you run test again? it seems to fail. Do you want I merge this for this version?

**Comment by Kukks** [comment]
tests pass

File: BTCPayServer/Configuration/ExternalConnectionString.cs [hunk 1]
```diff
@@ -77,6 +79,20 @@ public async Task<ExternalConnectionString> Expand(Uri absoluteUrlBase, External
...
+                {
+                    if (string.IsNullOrEmpty(connectionString.Password))
+                    {
+                        connectionString.Password = Guid.NewGuid().ToString();
+                    }
+
+                    await File.WriteAllTextAsync(connectionString.PasswordFile, connectionString.Password);
+                }
+            }
+
```
Comment (NicolasDorier) [inline BTCPayServer/Configuration/ExternalConnectionString.cs]: This should be called "CookieFile" and if "CookieFile", the value inside should not be cached, the file should be generated by the configurator, not by btcpayserver,

File: BTCPayServer/Controllers/ServerController.cs [hunk 2]
```diff
@@ -665,6 +668,15 @@ public async Task<IActionResult> Service(string serviceName, string cryptoCode,
                     case ExternalServiceTypes.LNDGRPC:
                     case ExternalServiceTypes.LNDRest:
                         return LndServices(service, connectionString, nonce);
+                    case ExternalServiceTypes.Configurator:
+                        
+                            return View("ConfiguratorService", new LightningWalletServices()
+                            {
+                                ShowQR = showQR,
+                                WalletName = service.ServiceName,
+                                ServiceLink = $"{connectionString.Server}?password={connectionString.Password}"
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/ServerController.cs]: The password will end up in the browser history.
The only way is to read the cookiefile, and the configurator to regenerate a new cookiefile once the value has been used. (RTL does that)

**Comment by Kukks** [comment]
updated @NicolasDorier

**Pull request merged**